src/Fmt_ast.ml

```diff
@@ -1099,8 +1105,11 @@ and fmt_expression c ?(box= true) ?eol ?parens ?ext ({ast= exp} as xexp) =
                          | Some xcnd ->
                              hvbox 0
                                ( hvbox 2
-                                   ( fmt_if (not first) "else " $ fmt "if "
-                                   $ fmt_expression c xcnd )
+                                   ( fmt_if (not first) "else "
+                                   $ ( if first then
+                                         fmt_keyword_with_extension "if" ext
+                                     else fmt "if" )
```

Comment:
I don't really understand the branch on `first` here. Is it desired to have
```
if%lwt e0 then e1 else if e2 then e3 else e4
```
instead of
```
if%lwt e0 then e1 else if%lwt e2 then e3 else e4
```
?
I guess my real concern is that `sugar_ite` may need to be extended to collect up the extensions.

Reply:
That is the intention - only having `%lwt` on the first `if`.  Now that I write that, though, I honestly don't know what the proper syntax would be.  I can check.

Reply:
I suspect that each ifthenelse expression can have an extension or not independently of the sub-expressions in the arms. In that case, `sugar_ite` will need to detect `Pexp_extension` super-expressions of `Pexp_ifthenelse` expressions, and return an additional optional extension with each `cnd` expression.